jmclient/jmclient/wallet_utils.py

```diff
@@ -1529,6 +1529,14 @@ def open_wallet(path, ask_for_password=True, password=None, read_only=False,
                         "you need to convert it using the conversion script "
                         "at `scripts/convert_old_wallet.py`".format(path))
 
+    if not read_only:
+        (lock_filename, locked_by_pid) = Storage.get_file_locking_status(path)
+        if locked_by_pid is not None:
+            raise Exception("File is currently in use (locked by pid {}). "
+                            "If this is a leftover from a crashed instance "
+                            "you need to remove the lock file `{}` manually.".
+                            format(locked_by_pid, lock_filename))
```

Comment:
We have to "raise exception" instead of "exit()" here, otherwise QT would quit when the file is locked.